### PR TITLE
require conmon v2.0.1

### PR DESCRIPTION
**Pull request by haircommander**
Signed-off-by: Peter Hunt <pehunt@redhat.com>

### CODE REVIEW

**Comment by mheon** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM, but some tests aren't hip

File: libpod/runtime.go [hunk 1]
```diff
@@ -89,6 +89,11 @@ var (
 	// DefaultDetachKeys is the default keys sequence for detaching a
 	// container
 	DefaultDetachKeys = "ctrl-p,ctrl-q"
+
+	// minConmonMajor is the major version required for conmon
+	minConmonMajor = 2
+	// minConmonVersion is a string used to display the minimum conmon version
+	minConmonVersion = "v2.0.0"
```
Comment (rhatdan) [inline libpod/runtime.go]: Can you make this
minConmonVersion = fmt.Sprintf("v%d.0.0", minConmonMajor)

Comment (rhatdan) [reply libpod/runtime.go]: Or better yet remove this variable and just use the minConmonMajor in the error message below.

Comment (haircommander) [reply libpod/runtime.go]: fixed

File: libpod/runtime.go [hunk 2]
```diff
@@ -829,7 +829,7 @@ func makeRuntime(ctx context.Context, runtime *Runtime) (err error) {
 
 	if !foundConmon {
 		if foundOutdatedConmon {
-			return errors.Wrapf(define.ErrConmonOutdated, "please update to v1.0.0 or later")
+			return errors.Wrapf(define.ErrConmonOutdated, "please update to v%s.0.0 or later", minConmonMajor)
```
Comment (rhatdan) [inline libpod/runtime.go]: Should be v%d.0.0

Comment (haircommander) [reply libpod/runtime.go]: oopsies

Comment (haircommander) [reply libpod/runtime.go]: golang is so forgiving...

**Comment by rhatdan** [comment]
/lgtm
/hold
Feel free to hold cancel when tests pass.

**Comment by mheon** [comment]
Large quantities of test failures - does this need a rebase?

**Comment by haircommander** [comment]
@mheon let's try that :D

**Comment by rhatdan** [comment]
Conmon is now in place on F30 and f31.

**Comment by vrothberg** [comment]
Do we need new images for the CI to make it pass?

`Error: could not get runtime: please update to v2.0.1 or later: outdated conmon version`

Parsing the logs, I found a minor nit: @haircommander, can we change the string to `outdated conmon version: please update to v2.0.1 or later`? Currently, we first print `please update to v2.0.1 or later` which confused me while reading as it didn't know what to update.

**Comment by haircommander** [comment]
@lsm5 did you use conmon v2.0.1 for 1.6-rc{1|2}? 
@vrothberg fixed the nit, and yeah we'd need v2.0.1 in the images for this to pass

**Comment by lsm5** [comment]
> @lsm5 did you use conmon v2.0.1 for 1.6-rc{1|2}?

I'm on it right now..

**Comment by rhatdan** [comment]
@haircommander @lsm5 This should be fixed now, correct?  Does this PR Need a rebase?

**Comment by haircommander** [comment]
@lsm5 has 1.6 made it into the PPAs? Ubuntu image build is failing cc: @cevich

**Comment by lsm5** [comment]
On Mon, Oct 07, 2019 at 06:43:04AM -0700, Peter Hunt wrote:
> @lsm5 has 1.6 made it into the PPAs? Ubuntu image build is failing cc: @cevich 

errr .. I had disabled the autobuild job for some reason (I guess to avoid
sending -rc to ppa).. I'll trigger it now ...
> 
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/containers/libpod/pull/3792#issuecomment-539017456

-- 
Lokesh
IRC, GitHub: lsm5
GPG: 0xC7C3A0DD
https://keybase.io/lsm5

**Comment by cevich** [comment]
@haircommander if you need to build VM images, might want to rebase onto or peek at #4189

**Comment by cevich** [comment]
@haircommander are we using new images here to pick up a new conmon?  (np if so, just checking)

**Comment by haircommander** [comment]
I just did the old tool chain, do we not need to now conmon is updated via dnf?

**Comment by cevich** [comment]
Nope you did the right thing, I was just curious if that was the reason and if current images don't contain a new enough conmon.  (I thought I rolled up new ones last week for this)

**Comment by haircommander** [comment]
@cevich good to know.

btw re: the in_podman problem. that'll have to be force merged through, right? then the post install will update the in_podman image and all will be well?

**Comment by rhatdan** [comment]
@haircommander Can we merge this now?  or does it need a rebase to merge?

**Comment by haircommander** [comment]
this is ready

**Comment by mheon** [comment]
Hung on a failed test

**Comment by mheon** [comment]
Restarted podman-in-podman but failures might be legitimate

**Comment by haircommander** [comment]
sorry folks, I think I missed the step of updating the dockerfile. I think in_podman should pass now?

**Comment by rhatdan** [comment]
/lgtm
/hold
hold cancel when tests pass.

**Comment by haircommander** [comment]
@cevich am I correct in saying that this PR needs to be merged before the changes in the dockerfile get used in the `in_podman` task? If that were the case, this would need to be force merged when all other tests pass

**Comment by haircommander** [comment]
manually pushed to the in_podman image. It should pass, then once we merge the post merge action will overwrite it organically. can I have a new lgtm @rhatdan @baude @vrothberg @TomSweeneyRedHat @mheon

**Comment by haircommander** [comment]
happy tests PTAL

File: Dockerfile [hunk 3]
```diff
@@ -57,7 +57,7 @@ RUN set -x \
 	&& rm -rf "$GOPATH"
 
 # Install conmon
-ENV CONMON_COMMIT 6f3572558b97bc60dd8f8c7f0807748e6ce2c440
+ENV CONMON_COMMIT 65fe0226d85b69fc9e527e376795c9791199153d
```
Comment (TomSweeneyRedHat) [inline Dockerfile]: This is fine for now, but consider changing this over  to the tag for the release like 'v2.0.2' then below I think (didn't try it) you can do `git checkout tags/CONMON_COMMIT`, at least that's the theory.

**Comment by TomSweeneyRedHat** [comment]
LGTM

File: Dockerfile (hunk 3 above)
Comment (haircommander) [reply Dockerfile]: I think actually the next step is to enable pulling conmon from a package (like the other vms do) instead of by git commit

**Comment by vrothberg** [comment]
Note that the code will slightly change as soon as I need to merge it into https://github.com/containers/libpod/pull/4352

**Pull request merged**